source/Fido.Utilities.pas

```diff
@@ -35,6 +36,10 @@   Utilities = record
     class function TryStringToTGuid(const Input: string; out Guid: TGuid): Boolean; static;
   end;
 
+  Guard = record
```

Comment:
If you put it inside Utilities we can avoid the name conflict.

Also, since you are there... Add also a CheckAssignedAndSet to use for the anonymous methods :)